torch_xla/core/xla_model.py

```diff
@@ -1080,15 +1081,17 @@ def select_fn(v):
   return ToXlaTensorArena(convert_fn, select_fn).transform(data)
 
 
-def send_cpu_data_to_device(data, device, input_sharding=None):
+def send_cpu_data_to_device(data,
+                            device,
+                            input_sharding: xs.ShardingSpec = None):
 
   def convert_fn(tensors):
     devices = [str(device)] * len(tensors)
     xtensors = torch_xla._XLAC._xla_tensors_from_aten(tensors, devices)
     if input_sharding:
-      assert(len(xtensors) == 2), \
-        f"Expected input data and label pair, but got {xtensors}."
-      input_sharding.apply(xtensors[0])
+      for xtensor in xtensors:
```

Comment:
can we add a unit test for this? or a e2e case that uses paraallel loader with SPMD.

Reply:
We have an e2e case using parallel loader in [test_train_spmd_imagenet.py](https://github.com/pytorch/xla/blob/master/test/spmd/test_train_spmd_imagenet.py#L255-L258)

Reply:
Yeah, this is the case with real imagenet data in the above script. We have an XL ML test which runs this configuration, and it's currently failing due to multiple (input, label) pairs since the existing implementation only accepts a single pair.